tsdb/db.go

```diff
@@ -740,6 +740,16 @@ func (db *DB) Compact() (err error) {
 	}()
 	// Check whether we have pending head blocks that are ready to be persisted.
 	// They have the highest priority.
+	checkpointTime := int64(math.MinInt64)
+	checkpointAttempted := false
+	defer func() {
+		if err != nil && !checkpointAttempted {
```

Comment:
If `db.stopc` is closed, no checkpoint is performed because of the check `err != nil`. Is it intended? Previous code would have run it.

Reply:
Good point! We need to truncate.

Reply:
s/truncate/checkpoint